mobile/mobile.go

```diff
@@ -174,16 +177,18 @@ func ConnectServer(nameSpace string, mailboxServer string, isDevServer bool,
 	}
 
 	mMutex.Lock()
-	defer mMutex.Unlock()
-
 	mc, ok := m[nameSpace]
+	mMutex.Unlock()
 	if !ok {
 		return fmt.Errorf("unknown namespace: %s", nameSpace)
 	}
 
 	// Since the connection function is blocking, we need to spin it off
 	// in another goroutine here. See https://pkg.go.dev/syscall/js#FuncOf.
 	go func() {
+		mcMutex.Lock()
```

Comment:
I'm not sure this changes the behavior that much. It's still a global lock, just narrower in its scope. If you want to have a lock per namespace, one way would be to have a map of mutexes (that itself is guarded by `mMutex`).

Reply:
@guggero Took this approach and pushed a change. LMK know what you think. Not sure if I have to explicitly check for each mutex here, but I included the check for now